kvbc/src/v4blockchain/v4_blockchain.cpp

```diff
@@ -581,6 +585,31 @@ std::optional<categorization::Value> KeyValueBlockchain::getLatest(const std::st
   return latest_keys_.getValue(category_id, key);
 }
 
+std::optional<categorization::Value> KeyValueBlockchain::getFromSnapshot(const std::string &category_id,
+                                                                         const std::string &key,
+                                                                         const BlockId version) const {
+  if (bftEngine::ReplicaConfig::instance().enableKeysHistoryCF) {
+    auto block_id = keys_history_.getVersionFromHistory(category_id, key, version);
+    if (!block_id) {
+      return std::nullopt;
+    }
+    return get(category_id, key, block_id.value());
+  } else {
+    BlockId limit = 0;
+    BlockId max_blocks_to_look = bftEngine::ReplicaConfig::instance().keysHistoryMaxBlocksNum;
```

Comment:
First we need to do getLatest from the latest CF to know the latest version of the key.
If it's higher than "version" then we should perform the iteration.

Reply:
Added the `getLatest` check.
but not exactly as it was in concord - I check for snapshot value also if `getLatest` returns nullopt, for the cases where we look for a key at snapshot x, and that key was deleted at block x+1 (or after).
btw, when key history CF is disabled, if bc returns nullopt for a key at a block, we can't know whether that key was deleted or just wasn't updated at the certain block. So our results are not accurate when we have a deleted key.
[that is not the case for the categorized storage, because its `getLatestVersion()` remember also deletes]

Reply:
Nice catch, add it as a comment, and we'll think whether it's worth fixing it